exercises/queen-attack/queen-attack.test.ts

```diff
@@ -0,0 +1,75 @@
+import Queens from './queen-attack'
+
+describe('Queens', () => {
+    it('has the correct default positions', () => {
+        const queens = new Queens()
+        expect(queens.white).toEqual([0, 3])
+        expect(queens.black).toEqual([7, 3])
+    })
+
+    xit('initialized with specific placement', () => {
+        const queens = new Queens({ white: [3, 7], black: [6, 1] })
+        expect(queens.white).toEqual([3, 7])
+        expect(queens.black).toEqual([6, 1])
+    })
+
+    xit('cannot occupy the same space', () => {
+        const positioning = { white: [2, 4], black: [2, 4] }
+        const expectedError = 'Queens cannot share the same space'
+        expect(() => new Queens(positioning)).toThrow(expectedError)
+    })
+
+    xit('toString representation', () => {
+        const positioning = { white: [2, 4], black: [6, 6] }
+        const queens = new Queens(positioning)
+        const board = ['_ _ _ _ _ _ _ _',
+            '_ _ _ _ _ _ _ _',
+            '_ _ _ _ W _ _ _',
+            '_ _ _ _ _ _ _ _',
+            '_ _ _ _ _ _ _ _',
+            '_ _ _ _ _ _ _ _',
+            '_ _ _ _ _ _ B _',
+            '_ _ _ _ _ _ _ _\n'].join('\n')
+        expect(queens.toString()).toEqual(board)
+    })
+
+    xit('queens cannot attack', () => {
+        const queens = new Queens({ white: [2, 3], black: [4, 7] })
```

Comment:
What happens if somebody provides an array that is too short  or too long?

Reply:
I just enforced them to be arrays of two `number` elements.